tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11156,60 +11156,66 @@ void Sema::DiagnoseReachableHLSLMethodCall(const CXXMethodDecl *MD,
     switch (opCode) {
     case hlsl::IntrinsicOp::MOP_CalculateLevelOfDetail:
     case hlsl::IntrinsicOp::MOP_CalculateLevelOfDetailUnclamped: {
-      if (!SM->IsSM68Plus()) {
-        QualType SamplerComparisonTy =
-            HLSLExternalSource::FromSema(this)->GetBasicKindType(
-                AR_OBJECT_SAMPLERCOMPARISON);
-        if (MD->getParamDecl(0)->getType() == SamplerComparisonTy) {
+      QualType SamplerComparisonTy =
+          HLSLExternalSource::FromSema(this)->GetBasicKindType(
+              AR_OBJECT_SAMPLERCOMPARISON);
+      if (MD->getParamDecl(0)->getType() == SamplerComparisonTy) {
+
+        if (!SM->IsSM68Plus()) {
+
           Diags.Report(Loc,
                        diag::warn_hlsl_intrinsic_overload_in_wrong_shader_model)
               << MD->getNameAsString() + " with SamplerComparisonState"
               << "6.8";
-        }
-      }
+        } else {
```

Comment:
I don't think this is quite what Tex was suggesting in the pseudo-code [here](https://github.com/microsoft/DirectXShaderCompiler/pull/6244#discussion_r1480848992). I think there was some meaningful whitespace missed. Here's how I interpretted it with braces for clarity: 
```C++
if < SM 6.8 {
  if new intrinsic {
    // emit diagnostic
  }
} else { // SM is 6.8+
   //do the rest of the diagnostics for shader stage and threadgroup dimensions
}
```

@tex3d can you confirm my undertanding?

Reply:
In conversation, I was corrected that the intent was to only produce these diagnostics on the new overloads so that old code can be compiled in 6.8 without any issue, so I withdraw my complaint. 

Nevermind I converted the wrong version of the pseudocode above 🤦